File: Course-Registration-System-main/java-(iteration_1)/src/LecturerReader.java
import java.util.ArrayList;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

import java.io.FileReader;
import java.io.IOException;

public class LecturerReader {
    public static ArrayList<Lecturer> readLecturersFromJsonFile(String filePath) {
        ArrayList<Lecturer> lecturers = new ArrayList<>();

        JSONParser jsonParser = new JSONParser();

        try (FileReader reader = new FileReader(filePath)) {
            // Parse the JSON file
            Object obj = jsonParser.parse(reader);

            // Cast the parsed object to JSONObject
            JSONObject jsonObject = (JSONObject) obj;

            // Get the array of lecturers
            JSONArray lecturerArray = (JSONArray) jsonObject.get("lecturers");

            // Iterate through the array and create Lecturer objects
            for (Object lecturerObj : lecturerArray) {
                JSONObject lecturerJson = (JSONObject) lecturerObj;

                // Extract information from the JSON object
                int id = ((Long) lecturerJson.get("id")).intValue();
                String password = (String) lecturerJson.get("password");
                String name = (String) lecturerJson.get("name");
                String surname = (String) lecturerJson.get("surname");
                String title = (String) lecturerJson.get("title");

                // Create a new Lecturer object and add it to the list
                Lecturer lecturer = new Lecturer(id, password, name, surname, title);
                lecturers.add(lecturer);
            }

        } catch (IOException | ParseException e) {
            e.printStackTrace(); // Handle the exception according to your needs
        }

        return lecturers;
    }
}

File: Course-Registration-System-main/python-(iteration_2)/src/LecturerReader.py
import json
from Lecturer import Lecturer

def readLecturersFromJsonFile(file_path):
    lecturers = []
    try:
        with open(file_path, 'r') as file:
            # Parse the JSON file
            data = json.load(file)

            # Get the array of lecturers
            lecturer_array = data.get("lecturers", [])

            # Iterate through the array and create Lecturer objects
            for lecturer_json in lecturer_array:
                # Extract information from the JSON object
                lecturer_id = int(lecturer_json.get("id", 0))
                password = lecturer_json.get("password", "")
                name = lecturer_json.get("name", "")
                surname = lecturer_json.get("surname", "")
                title = lecturer_json.get("title", "")

                # Create a new Lecturer object and add it to the list
                lecturer = Lecturer(lecturer_id, password, name, surname, title)
                lecturers.append(lecturer)

    except (IOError, json.JSONDecodeError) as e:
        print(f"Error: {e}")
    return lecturers
